src/ts/components/seekbarlabel.ts

```diff
@@ -136,31 +136,50 @@ export class SeekBarLabel extends Container<SeekBarLabelConfig> {
       // We use the thumbnail image loader to make sure the thumbnail is loaded and it's size is known before be can
       // calculate the CSS properties and set them on the element.
       this.thumbnailImageLoader.load(thumbnail.url, (url, width, height) => {
-        let thumbnailCountX = width / thumbnail.w;
-        let thumbnailCountY = height / thumbnail.h;
-
-        let thumbnailIndexX = thumbnail.x / thumbnail.w;
-        let thumbnailIndexY = thumbnail.y / thumbnail.h;
-
-        let sizeX = 100 * thumbnailCountX;
-        let sizeY = 100 * thumbnailCountY;
-
-        let offsetX = 100 * thumbnailIndexX;
-        let offsetY = 100 * thumbnailIndexY;
-
-        let aspectRatio = 1 / thumbnail.w * thumbnail.h;
-
-        // The thumbnail size is set by setting the CSS 'width' and 'padding-bottom' properties. 'padding-bottom' is
-        // used because it is relative to the width and can be used to set the aspect ratio of the thumbnail.
-        // A default value for width is set in the stylesheet and can be overwritten from there or anywhere else.
-        thumbnailElement.css({
-          'display': 'inherit',
-          'background-image': `url(${thumbnail.url})`,
-          'padding-bottom': `${100 * aspectRatio}%`,
-          'background-size': `${sizeX}% ${sizeY}%`,
-          'background-position': `-${offsetX}% -${offsetY}%`,
-        });
+        if (['x', 'y', 'w', 'h'].every((key) => thumbnail.hasOwnProperty(key))) {
```

Comment:
That could have been much simpler and faster (no function calls):

```ts
if (thumbnail.x === undefined) {
  // Thumbnails without x/y/w/h values are single-image thumbnails, not sprites within an image
  // We set these values to simulate a single-sprite image whose sprite covers the whole image
  thumbnail.x = 0;
  thumbnail.y = 0;
  thumbnail.w = width;
  thumbnail.h = height;
}
```

Reply:
I'd prefer the done approach as for single images we need no `background-position` and can also omit the calculations.

Reply:
Then please simplify the check to `if (thumbnail.x === undefined) ...`